auto.js

```diff
@@ -563,34 +569,120 @@ var CookieAuto = {};
                 for (let o of u) {
                     let x = o.icon[0]*-48;
                     let y = o.icon[1]*-48;
-                    res += '<div class="icon" onclick="CookieAuto.toggleInShoppingList(Game.UpgradesById['+o.id+']);" onmouseout="Game.setOnCrate(0);Game.tooltip.shouldHide=1;" onmouseover="if (!Game.mouseDown) {Game.setOnCrate(this);Game.tooltip.dynamic=1;Game.tooltip.draw(this,function(){return function(){return Game.crateTooltip(Game.UpgradesById['+o.id+'],\'shoppingListSelector\');}();},\'\');Game.tooltip.wobble();}" style="padding:0; cursor:pointer; background-position:'+x+'px '+y+'px; float:left; opacity:'+(inShoppingList(o)?'1':'0.2')+';"></div>';
+                    res += '<div id="shlst'+o.id+'" class="icon" onclick="CookieAuto.toggleInShoppingList(Game.UpgradesById['+o.id+']);" onmouseout="Game.setOnCrate(0);Game.tooltip.shouldHide=1;" onmouseover="if (!Game.mouseDown) {Game.setOnCrate(this);Game.tooltip.dynamic=1;Game.tooltip.draw(this,function(){return function(){return Game.crateTooltip(Game.UpgradesById['+o.id+'],\'shoppingListSelector\');}();},\'\');Game.tooltip.wobble();}" style="padding:0; cursor:pointer; background-position:'+x+'px '+y+'px; float:left; opacity:'+(inShoppingList(o)?'1':'0.2')+';"></div>';
                 }
                 return res;
+            },
+
+            ui : {
+                showing : false,
+                menuelt : null,
+                init : function () {
+                    this.menuelt = document.createElement('div');
+                    this.menuelt.style.position = "absolute";
+                    this.menuelt.style.left = "calc(30% + 16px)";
+                    this.menuelt.style.top = "144px";
+                    this.menuelt.style.width = "calc(70vw - 348px)";
+                    this.menuelt.style.height = "calc(100vh - 144px)";
+                    this.menuelt.style.wordWrap = "break-word";
+                    this.menuelt.style.zIndex = "10";
+                    this.menuelt.style.background = "#000 url(img/darkNoise.jpg)";
+                    this.menuelt.style.overflow = "auto";
+                    this.menuelt.style.display = "none";
+
+                    document.body.appendChild(this.menuelt);
+
+                    Game.prefs.buyscript_abbuild = CookieAuto.control.buyBuildings?1:0;
```

Comment:
I know we need CookieAuto references in HTML, but for cases like this, replace CookieAuto.control with settings

Reply:
done